OPHD/UI/Core/RadioButton.cpp

```diff
@@ -1,30 +1,23 @@
-#include "RadioButton.h"
-
-#include "UIContainer.h"
-#include "../../Cache.h"
-#include "../../Constants.h"
-
-#include <NAS2D/Utility.h>
-#include <NAS2D/Renderer/Renderer.h>
-#include <NAS2D/MathUtils.h>
-
-#include <algorithm>
-
+#include "RadioButtonGroup.h"
 
 using namespace NAS2D;
 
-
-RadioButton::RadioButton(std::string newText) :
+RadioButtonGroup::RadioButton::RadioButton(std::string newText, RadioButtonGroup* parentContainer, NAS2D::DelegateX<void> delegate) :
 	mFont{fontCache.load(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL)},
-	mSkin{imageCache.load("ui/skin/checkbox.png")},
-	mLabel{newText}
+	mSkin{imageCache.load("ui/skin/radio.png")},
+	mLabel{newText},
+	mParentContainer{parentContainer}
 {
-	Utility<EventHandler>::get().mouseButtonDown().connect(this, &RadioButton::onMouseDown);
+	hasFocus(true);
```

Comment:
This `hasFocus` call stands out as looking a bit strange and unexpected. Should this perhaps be handled by the `RadioButtonGroup` when an `index` is selected?

Reply:
I tried adding it in the setSelected method, but then the clicks don't register.
I found that placing it in `update()` also works. But I don't know if that is any better.

Reply:
What is the purpose of the `hasFocus` call? I'm thinking it allows a `Control` to be highlighted so the user knows where input will go. I see tracking the active control as being relevant only to keyboard input. I don't know why it would be called on every `RadioButton`. I'm not super familiar with how it's used in the API though.

Reply:
I have looked into where the hasFocus() call is being used to apply said information, and I see there is a total of 2 checks in  `Window.cpp`:

```
void Window::onMouseDown(EventHandler::MouseButton button, int x, int y)
{
	if (!visible() || !hasFocus()) { return; }

	UIContainer::onMouseDown(button, x, y);

	const auto titleBarBounds = NAS2D::Rectangle{mRect.x, mRect.y, mRect.width, sWindowTitleBarHeight};
	mMouseDrag = (button == EventHandler::MouseButton::Left && titleBarBounds.contains(NAS2D::Point{x, y}));
}

```

Could it be that this is blocking mouse clicks when its focus is set to `false`?

![image](https://user-images.githubusercontent.com/3091512/118177443-fbc09980-b432-11eb-8f83-c5c3cf760d36.png)

Reply:
Ahh, yes, good catch. That is very strange to be checking in a `onMouseDown` event handler. Typically "focus" is only related to keyboard input, not mouse input.

I would recommend removing the `!hasFocus()` check from `onMouseDown`.

@ldicker83 : Did you want to weight in on that? You might be more familiar with why that check was added in the first place.

Reply:
I tried checking `git blame` to see the history of that line. Though it seems it was just copied from other files, and use of `hasFocus` in mouse event handlers is quite widespread in the code base. I opened #884 to track.

I suppose we can mark this as "resolved", and worry about it as part of the other issue. It would make sense to address that issue for all controls simultaneously.